apps/sms/views/shared/js/settings.js

```diff
@@ -35,14 +36,16 @@ var Settings = {
     var keyHandlerSet = {
       'dom.mms.operatorSizeLimitation': this.initMmsSizeLimitation.bind(this),
       'operatorResource.sms.maxConcat':
-        this.initSmsMaxConcatenatedMsg.bind(this)
+        this.initSmsMaxConcatenatedMsg.bind(this),
+      'messages.mms.sendReadReport.enabled':
+        this.initSendReadReport.bind(this)
     };
     var settings = navigator.mozSettings;
     var conns = navigator.mozMobileConnections;
 
     function setHandlerMap(key) {
       var req = settings.createLock().get(key);
-      req.onsuccess = function settings_getSizeSuccess() {
+      req.onsuccess = function settings_createLockSuccess() {
```

Comment:
@julienw  didn't change to .then(), as it is effecting the createLock.get(key) mock. So, didn't get better option to resolve this. If you can help me here.?